baseapp/baseapp.go

```diff
@@ -398,6 +485,14 @@ func (app *BaseApp) runTx(isCheckTx bool, txBytes []byte, tx sdk.Tx) (result sdk
 		ctx = app.deliverState.ctx.WithTxBytes(txBytes)
 	}
 
+	// Create a new zeroed gas meter
+	ctx = ctx.WithGasMeter(sdk.NewGasMeter(app.txGasLimit))
```

Comment:
doesn't the gas meter need to use the gas provided in the tx, rather than this app level limit ? Perhaps we need to do this set up in the anteHandler

Reply:
Yes, this is wrong, thanks. Switched to ante handler.

Do we also need to check that `tx.Fee.Gas <= app.txGasLimit`, or will Tendermint do that for us?

Reply:
we'll need to https://github.com/cosmos/cosmos-sdk/issues/1007